### PR TITLE
enhance: pack mods & key into a u32 as id instead of hashing

### LINKED ISSUE #101: Hash for hotkey id is unnecessary
```rust
pub fn new(mods: Option<Modifiers>, key: Code) -> Self {
        let mut mods = mods.unwrap_or_else(Modifiers::empty);
        if mods.contains(Modifiers::META) {
            mods.remove(Modifiers::META);
            mods.insert(Modifiers::SUPER);
        }

        let mut hotkey = Self { mods, key, id: 0 };
        hotkey.id = hotkey.generate_hash();
        hotkey
    }

    fn generate_hash(&self) -> u32 {
        let hotkey_str = self.into_string();
        let mut hasher = std::collections::hash_map::DefaultHasher::new();
        hotkey_str.hash(&mut hasher);
        std::hash::Hasher::finish(&hasher) as u32
    }
```
    
I'm curious why a hash is being used here rather than just leaving the base u32 value in, it's a guaranteed unique value + it'll always be 32 bits. Am I missing something? 

i.e.

it should rather be
```rust
pub fn new(mods: Option<Modifiers>, key: Code) -> Self {
        let mut mods = mods.unwrap_or_else(Modifiers::empty);
        if mods.contains(Modifiers::META) {
            mods.remove(Modifiers::META);
            mods.insert(Modifiers::SUPER);
        }

        let mut hotkey = Self { mods, key, id: 0 };
        hotkey.id = mods as u32;
        hotkey
    }
```

If the id is related to modifiers I can just match on the concatenation of the flags again after throughout the codebase. Wouldn't mind opening a pr.

### PR DESCRIPTION
**Pull request by mixy1**
Change mentioned in #101.

### CODE REVIEW

File: src/hotkey.rs [hunk 1]
```diff
@@ -92,20 +92,13 @@ impl HotKey {
         }
 
         let mut hotkey = Self { mods, key, id: 0 };
-        hotkey.id = hotkey.generate_hash();
+        hotkey.id = (mods.bits() as u64) << 32 | key as u64;
         hotkey
```
Comment (amrbashir) [inline src/hotkey.rs:96]: On a second thought, I think we can fit the whole id into a u32, as `Code` only contains a 215 variant, so the maximum possible value is `214` which is no where near the max value of a u16 (65535), same can be said for modifiers. 
```rs
        hotkey.id = mods.bits() << 16 | key;
```

We can also simplify this code a bit to be 

```suggestion
        Self { mods, key, id: mods.bits() << 16 | key }
```

**Review by amrbashir — APPROVED** [review]

File: src/hotkey.rs (hunk 1 above)
Comment (amrbashir) [inline src/hotkey.rs:96]: On a second thought, I think we can fit the whole id into a u32, as `Code` only contains a 215 variant, so the maximum possible value is `214` which is no where near the max value of a u16 (65535), same can be said for modifiers. 
```rs
        hotkey.id = mods.bits() << 16 | key;
```

We can also simplify this code a bit to be 

```suggestion
        Self { mods, key, id: mods.bits() << 16 | key }
```

Comment (amrbashir) [inline src/hotkey.rs]: On a second thought, I think we can fit the whole id into a u32, as `Code` only contains a 215 variant, so the maximum possible value is `214` which is no where near the max value of a u16 (65535), same thing for modifiers. 
```rs
        hotkey.id = mods.bits() << 16 | key;
```

We can also simplify this code a bit to be 

```suggestion
        Self { mods, key, id: mods.bits() << 16 | key }
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Comment by mixy1** [comment]
Yeah seems more reasonable to use 16 bits here for each value.

Amended as requested, interesting note, I found a small discrepancy on the rust docs as well in the process: https://doc.rust-lang.org/reference/attributes/type_system.html. (Code is non_exhaustive, yet this runs perfectly well, I've opened a pr for that too)

**Review by amrbashir — COMMENTED** [review]
Just clippy and fmt fixes and this should be good to go, thanks

**Comment by mixy1** [comment]
Should be fine now. Thanks!

**Review by amrbashir — APPROVED** [review]

**Pull request merged**